[DO NOT MERGE] cli: add OCI to image save/load

runcom opened the pull request:
Client side stuff split from moby/moby#26369

/cc @stevvooe @thaJeztah @cpuguy83 @vbatts 

server side is docker/docker#33355 

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

runcom commented:
This PR vendors https://github.com/moby/moby/pull/33355 to build the binary, so don't merge it yet

stevvooe commented:
@runcom Do you have time to update this with a workflow description? :)

runcom commented:
Where do I have to add a workflow description? The commands have been designed/discussed here mostly https://github.com/moby/moby/pull/26369

stevvooe commented:
@runcom Just to the description of this PR. If they are the same, just copy them over so people don't have to guess.

I'll still build this version and give it a try in bit.

thaJeztah commented:
@stevvooe have you been able to give this one a go?

also ping @runcom for the updated description :smile:

stevvooe commented:
@thaJeztah The other PR needs to be rebased but we haven't heard back from @runcom. I am wondering if we are going to have to carry these PRs.

runcom commented:
I have fixed, rebased and added tests on the Moby side. I'd love to have that merged first before diving into docker itself. I'll rebase this as I get a chance though.
